ext/readline/readline.c

```diff
@@ -183,11 +183,24 @@ PHP_FUNCTION(readline_info)
 		if (zend_string_equals_literal_ci(what,"line_buffer")) {
 			oldstr = rl_line_buffer;
 			if (value) {
-				/* XXX if (rl_line_buffer) free(rl_line_buffer); */
 				if (!try_convert_to_string(value)) {
 					RETURN_THROWS();
 				}
-				rl_line_buffer = strdup(Z_STRVAL_P(value));
+#ifndef PHP_WIN32
+				if (strlen(oldstr) < Z_STRLEN_P(value)) {
+					rl_extend_line_buffer(Z_STRLEN_P(value) + 1);
+				}
+				strcpy(rl_line_buffer, Z_STRVAL_P(value));
```

Comment:
Why not memcpy? We already checked that the buffer is large enough

Reply:
that s the thing, since we know it s large enough strcpy fits by its simplicity :)

Reply:
after 2nd thoughts, will give it a try.